docs/IR.md

```diff
@@ -1,127 +1,372 @@
 
 #### Values
 
-The representation distinguishes between two kinds of values: attribute values, which are statically known, and runtime values. The type of values permitted in the two cases are different. The permitted types of attribute values are indicated by the enumeration `AttributeType`, while the permitted types of runtime values are described by `TypeProto`. 
+The representation distinguishes between two kinds of values: attribute values, which are statically known, and runtime values. The types of values permitted in the two cases are different.
 
-The types of the inputs and outputs of the model must be specified, including the shapes of tensors. While the ranks of input and output tensors are statically specified, the sizes of specific dimensions (axis) may be statically unknown and are indicated so using symbolic identifiers in the shape. 
+The types of the inputs and outputs of a graph must be specified.
 
-#### Optional Inputs
+#### Variadic Inputs and Outputs
+ 
+The last input or output of an operator MAY be marked as variadic. For example, the operator 'Max()' can be used to compute the maximum of a varying number of input values.
 
-Some Operators have inputs that are marked as optional. There are two ways to leave an optional input unspecified. The first is to simply not provide that input. However, this is not always possible - for example, if you wish to leave the fourth input unspecified, but still provide a value for the fifth input. Therefore, any input with a name of the empty string is treated as an unspecified optional input.
+For each variadic operator input, one or more node inputs must be specified. For each variadic operator output, one or more node outputs must be specified. 
 
-Built-in Operators and Standard Data Types
-------------------------------------------
+#### Optional Inputs and Outputs
 
-### Operators
+Some operators have inputs that are marked as optional, which means that a referring node MAY choose not to provide a value for that input.
+
+Some operators have outputs that are optional, which means that an operator, depending on its input parameters and/or attributes, MAY choose not to compute all its declared outputs. 
+
+There are two ways to leave an optional input or output unspecified: the first, available only for trailing inputs and outputs, is to simply not provide that input; the second method is to use an empty string in place of an input or output name.
+
+Each node referring to an operator with optional outputs MUST provide a name for each output that is computed and MUST NOT provide names for outputs that are not computed.
+
+
+## Standard data types
+
+There are two official ONNX variants; the main distinction between the two is found in the supported types. As mentioned earlier, the __ONNX__ definition recognizes only tensors as input and output types, while the Classical Machine Learning extension. __ONNX-ML__, also recognizes sequences and maps.
```

Comment:
I believe they also support different operator sets. For example, "LinearClassifier" is not a part of ONNX but belongs to ONNX-ML. Please correct me if I am wrong. Thanks!

Reply:
Yes, that is an important point. Thanks!